assets/helpers/__tests__/utilitiesTest.js

```diff
@@ -59,4 +75,88 @@ describe('utilities', () => {
 
   });
 
+  describe('roundDp', () => {
+
+    it('should by default round to two decimal places', () => {
+      expect(roundDp(1234.5678)).toBe(1234.57);
+    });
+
+    it('should round, not floor or ceil', () => {
+      expect(roundDp(12.345)).toBe(12.35);
+      expect(roundDp(12.344)).toBe(12.34);
+    });
+
+    it('should round to a given number of decimal places', () => {
+      expect(roundDp(12.3456789, 5)).toBe(12.34568);
+    });
+
+  });
+
+  describe('generateClassName', () => {
+
+    it('should create the same classname if no modifier is passed', () => {
+      expect(generateClassName('made-up-class')).toBe('made-up-class');
+    });
+
+    it('should return a classname with a modifier attached', () => {
+      expect(generateClassName('made-up-class', 'fake-modifier'))
+        .toBe('made-up-class made-up-class--fake-modifier');
+    });
+
+  });
+
+  describe('clickSubstituteKeyPressHandler', () => {
+
+    it('should call callback if return is pressed', (done) => {
+      clickSubstituteKeyPressHandler(done)(getMockedKeypress(13));
+    });
+
+    it('should call callback if space is pressed', (done) => {
+      clickSubstituteKeyPressHandler(done)(getMockedKeypress(32));
+    });
+
+  });
+
+  describe('parseBoolean', () => {
+
+    it('should parse all variations of true', () => {
+      expect(parseBoolean('true', false)).toBe(true);
+      expect(parseBoolean('TRUE', false)).toBe(true);
+      expect(parseBoolean('True', false)).toBe(true);
+    });
+
+    it('should parse all variations of false', () => {
+      expect(parseBoolean('false', true)).toBe(false);
+      expect(parseBoolean('FALSE', true)).toBe(false);
+      expect(parseBoolean('False', true)).toBe(false);
+    });
+
+    it('should produce the correct fallbacks on failure to parse', () => {
+      expect(parseBoolean('notaboolean', true)).toBe(true);
+      expect(parseBoolean('notaboolean', false)).toBe(false);
+    });
+
+  });
+
+  describe('deserialiseJson', () => {
+
+    it('should deserialise a valid JSON object', () => {
+
+      const serialised = '{"a": 1, "b": "hello world"}';
+      const deserialised = { a: 1, b: 'hello world' };
+
+      expect(deserialiseJson(serialised)).toEqual(deserialised);
+
+    });
+
+    it('should return null for JSON that is malformed', () => {
+
+      const serialised = '{{notvalidJSON';
+
+      expect(deserialiseJson(serialised)).toBeNull();
```

Comment:
This `null` is returned after `JSON.parse` fails to parse the proposed JSON. Since `JSON.parse` always return an object or throws an error, Is it possible to delete the validation done on line 67? As far as I understand, there is no case where line 71 will be executed.

Reply:
`JSON.parse` doesn't always return an object, the following is `true`:

```
typeof JSON.parse('"hello"') !== 'object'
```

However, I thought I'd removed the object check, I meant to when I renamed it to `deserialiseJson`, my bad =S. It's not helpful because other things like arrays come back as objects.